src/diff.c

```diff
@@ -580,14 +580,15 @@ int git_diff__oid_for_file(
 	entry.file_size = size;
 	entry.path = (char *)path;
 
-	return git_diff__oid_for_entry(out, diff, &entry, NULL);
+	return git_diff__oid_for_entry(out, diff, &entry, NULL, NULL);
 }
 
 int git_diff__oid_for_entry(
 	git_oid *out,
 	git_diff *diff,
 	const git_index_entry *src,
-	const git_oid *update_match)
+	const git_oid *update_match,
+	bool *index_updated)
```

Comment:
We put `out`-type parameters as the first params - could you change this to `git_diff__oid_for_entry(git_oid *out, bool *index_updated...`